src/app/modules/vault-filter/components/folder-filter.component.html

```diff
@@ -0,0 +1,87 @@
+<ng-container *ngIf="!hide">
+  <div class="filter-heading">
+    <button
+      class="toggle-button"
+      [attr.aria-expanded]="!isCollapsed(foldersGrouping)"
+      appStopProp
+      (click)="toggleCollapse(foldersGrouping)"
+      aria-expanded="!isCollapsed(foldersGrouping)"
+    >
+      <i
+        class="bwi bwi-fw"
+        title="{{ 'toggleCollapse' | i18n }}"
+        aria-hidden="true"
+        [ngClass]="{
+          'bwi-angle-right': isCollapsed(foldersGrouping),
+          'bwi-angle-down': !isCollapsed(foldersGrouping)
+        }"
+      ></i>
+    </button>
+    <h2>&nbsp;{{ foldersGrouping.name | i18n }}</h2>
+    <button
+      class="add-button"
+      appBlurClick
+      (click)="addFolder()"
+      appA11yTitle="{{ 'addFolder' | i18n }}"
+    >
+      <i class="bwi bwi-plus bwi-fw" aria-hidden="true"></i>
+    </button>
+  </div>
+  <ul class="filter-options" *ngIf="!isCollapsed(foldersGrouping)">
+    <ng-template #recursiveFolders let-folders>
+      <li
+        *ngFor="let f of folders"
+        [ngClass]="{
+          active: f.node.id === activeFilter.selectedFolderId && activeFilter.selectedFolder
+        }"
+        class="filter-option"
+      >
+        <span class="filter-buttons">
+          <button class="toggle-button" *ngIf="f.children.length" (click)="toggleCollapse(f.node)">
+            <i
+              class="bwi bwi-fw"
+              title="{{ 'toggleCollapse' | i18n }}"
+              aria-hidden="true"
+              [ngClass]="{
+                'bwi-angle-right': isCollapsed(f.node),
+                'bwi-angle-down': !isCollapsed(f.node)
+              }"
+              (click)="toggleCollapse(f.node)"
+              appStopProp
+            ></i>
+          </button>
+          <button class="filter-button" appStopClick appBlurClick (click)="applyFilter(f.node)">
+            <i *ngIf="f.children.length === 0" class="bwi bwi-fw bwi-folder" aria-hidden="true"></i>
+            &nbsp;{{ f.node.name }}
+          </button>
+          <button
+            class="edit-button"
+            *ngIf="f.node.id"
+            appStopClick
+            appBlurClick
+            (click)="editFolder(f.node)"
+            appA11yTitle="{{ 'editFolder' | i18n }}"
+          >
+            <i class="bwi bwi-pencil bwi-fw" aria-hidden="true"></i>
+            <span
```

Comment:
I don't understand what this `<span>` is doing here - it just seems to duplicate the `<button>` el, but it's otherwise empty.

Reply:
[[bug] Remove empty span element](https://github.com/bitwarden/desktop/pull/1487/commits/97d4aea9e6301004000727a7373e9df0b34ee857)